src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -199,4 +215,71 @@ public TextExerciseResource(TextExerciseRepository textExerciseRepository, UserS
         }
         return ResponseEntity.notFound().build();
     }
+
+    /**
+     * Returns the data needed for the text editor, which includes the participation, textSubmission with model if existing
+     * and the assessments if the submission was already submitted.
+     *
+     * @param participationId the participationId for which to find the data for the text editor
+     * @return the ResponseEntity with json as body
+     */
+    @GetMapping("/text-editor/{participationId}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    @Timed
+    public ResponseEntity<JsonNode> getDataForTextEditor(@PathVariable Long participationId) {
+        Participation participation = participationService.findOne(participationId);
+        if (participation == null) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "participationNotFound", "No participation was found for the given ID.")).body(null);
+        }
+        TextExercise textExercise;
+        if (participation.getExercise() instanceof TextExercise) {
+            textExercise = (TextExercise) participation.getExercise();
+            if (textExercise == null) {
+                return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("textExercise", "exerciseEmpty", "The exercise belonging to the participation is null.")).body(null);
+            }
+        } else {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("textExercise", "wrongExerciseType", "The exercise of the participation is not a modeling exercise.")).body(null);
+        }
+
+        // users can only see their own submission (to prevent cheating), TAs, instructors and admins can see all models
+        if (!authCheckService.isOwnerOfParticipation(participation) && !courseService.userHasAtLeastTAPermissions(textExercise.getCourse())) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
+
+        // if no results, check if there are really no results or the relation to results was not updated yet
+        if (participation.getResults().size() <= 0) {
+            List<Result> results = resultRepository.findByParticipationIdOrderByCompletionDateDesc(participation.getId());
+            participation.setResults(new HashSet<>(results));
+        }
+
+        ObjectNode data = objectMapper.createObjectNode();
+
+        TextSubmission textSubmission = participation.findLatestTextSubmission();
+        // NOTE: avoid infinite recursion by setting submissions to null
+        participation.setSubmissions(null);
+        data.set("participation", objectMapper.valueToTree(participation));
```

Comment:
This Endpoint exposes the Sample Solution:
```
{
  "participation" : {
    "id" : 1,
    "repositoryUrl" : null,
    "buildPlanId" : null,
    "initializationState" : "INITIALIZED",
    "initializationDate" : "2018-11-07T13:07:38+01:00",
    "presentationScore" : null,
    "results" : [ ],
    "submissions" : null,
    "student" : {
      "id" : 1,
      "login" : "ga97kaw",
      "firstName" : "Jan Philip Bernius",
      "lastName" : "",
      "email" : "janphilip.bernius@tum.de",
      "activated" : true,
      "langKey" : "en",
      "imageUrl" : null,
      "resetDate" : null,
      "groups" : null
    },
    "exercise" : {
      "type" : "text",
      "id" : 1,
      "title" : "Text Exercise",
      "shortName" : null,
      "releaseDate" : null,
      "dueDate" : null,
      "maxScore" : 5.0,
      "problemStatement" : "Das ist ein Test.",
      "gradingInstructions" : null,
      "categories" : null,
      "difficulty" : null,
      "course" : {
        "id" : 1,
        "title" : "ArTEMiS",
        "shortName" : null,
        "studentGroupName" : "tumuser",
        "teachingAssistantGroupName" : "artemis-dev",
        "instructorGroupName" : "ls1doktoranden",
        "startDate" : "2018-11-03T23:00:52+01:00",
        "endDate" : "2018-12-31T22:59:02+01:00",
        "onlineCourse" : false
      },
      "sampleSolution" : "Foo Bar",                // <- THIS!
      "visibleToStudents" : true,
      "ended" : false
    },
    "repositoryUrlAsUrl" : null
  }
}
```

We should remove it from the response!

Maybe like that:
```suggestion
        ((TextExercise) participation.getExercise()).setSampleSolution(null);
        data.set("participation", objectMapper.valueToTree(participation));
```